core/pkg/resourcehandler/k8sresources.go

```diff
@@ -133,25 +143,62 @@ func (k8sHandler *K8sResourceHandler) GetResources(sessionObj *cautils.OPASessio
 
 	// check that controls use cloud resources
 	if len(cloudResources) > 0 {
-		provider, err := getCloudProviderDescription(allResources, ksResourceMap)
+		err := k8sHandler.collectCloudResources(sessionObj, allResources, ksResourceMap, cloudResources)
 		if err != nil {
-			cautils.SetInfoMapForResources(err.Error(), cloudResources, sessionObj.InfoMap)
-			logger.L().Warning("failed to collect cloud data", helpers.Error(err))
+			logger.L().Warning("failed to collect cloud resources", helpers.Error(err))
 		}
-		if provider != "" {
-			if sessionObj.Metadata != nil && sessionObj.Metadata.ContextMetadata.ClusterContextMetadata != nil {
-				sessionObj.Metadata.ContextMetadata.ClusterContextMetadata.CloudProvider = provider
+	}
+
+	return k8sResourcesMap, allResources, ksResourceMap, nil
+}
+
+func (k8sHandler *K8sResourceHandler) collectCloudResources(sessionObj *cautils.OPASessionObj, allResources map[string]workloadinterface.IMetadata, ksResourceMap *cautils.KSResources, cloudResources []string) error {
+	var err error
+	clusterName := cautils.ClusterName
+	provider := cloudsupport.GetCloudProvider(clusterName)
```

Comment:
[TODO]: I think we get the cloud provider in a different place in the code. Please verify this.

Reply:
I moved it to here